invoice/src/main/java/org/killbill/billing/invoice/dao/DefaultInvoiceDao.java

```diff
@@ -854,4 +859,47 @@ private void validateInvoiceItemToBeAdjusted(final InvoiceItemSqlDao invoiceItem
         }
     }
 
+    @Override
+    public void changeInvoiceStatus(final UUID invoiceId, final InvoiceStatus newStatus,
+                                    final InternalCallContext context) throws InvoiceApiException {
+        transactionalSqlDao.execute(InvoiceApiException.class, new EntitySqlDaoTransactionWrapper<Void>() {
+            @Override
+            public Void inTransaction(final EntitySqlDaoWrapperFactory entitySqlDaoWrapperFactory) throws Exception {
+                final InvoiceSqlDao transactional = entitySqlDaoWrapperFactory.become(InvoiceSqlDao.class);
+
+                // Retrieve the invoice and make sure it belongs to the right account
+                final InvoiceModelDao invoice = transactional.getById(invoiceId.toString(), context);
+
+                if (invoice == null ) {
+                    throw new InvoiceApiException(ErrorCode.INVOICE_NOT_FOUND, invoiceId);
+                }
+
+                if (invoice.getStatus().equals(newStatus)) {
+                    throw new InvoiceApiException(ErrorCode.INVOICE_INVALID_STATUS, newStatus, invoiceId, invoice.getStatus());
+                }
+
+                transactional.updateStatus(invoiceId.toString(), newStatus.toString(), context);
+
+                // notify invoice creation event
+                final BigDecimal balance = InvoiceModelDaoHelper.getBalance(invoice);
+                final DefaultInvoiceCreationEvent defaultInvoiceCreationEvent = new DefaultInvoiceCreationEvent(invoice.getId(), invoice.getAccountId(),
+                                                                                                                balance, invoice.getCurrency(),
+                                                                                                                context.getAccountRecordId(), context.getTenantRecordId(),
+                                                                                                                context.getUserToken());
+                postEvent(defaultInvoiceCreationEvent, invoice.getAccountId());
+
+                return null;
+            }
+        });
+    }
+
+    /* Use this method to post any event that implements BusInternalEvent */
+    private void postEvent(final BusInternalEvent event, final UUID accountId) {
+        try {
+            eventBus.post(event);
```

Comment:
You can call your method `notifyBusOfInvoiceCreation` to make it similar to the one that already exists (`notifyBusOfInvoiceAdjustment`)